## Problem Statement 1:

Blood glucose levels for obese patients have a mean of 100 with a standard deviation of 15.A researcher thinks that a diet high in raw cornstarch will have a positive effect on blood glucose levels. A sample of 36 patients who have tried the raw cornstarch diet have a mean glucose level of 108. Test the hypothesis that the raw cornstarch had an effect or not.

In [1]:
import numpy as np

null hypothesis and the alternate hypothesis

H0 : mu = 100    
H1 : mu != 100

In [2]:
def Ztest(s_mean , p_mean , n , std):
    return (s_mean - p_mean)/(std / np.sqrt(n))

In [3]:
s_mean = 108
p_mean = 100
n = 36
std = 15
z_value = Ztest(s_mean , p_mean , n , std)
print("Z value :",z_value)

Z value : 3.2


We use significance level α = 0.05, but because of a two-sided test we have two rejection regions. They are z < −1.96 or z > 1.96..

In [4]:
if z_value > 1.96 or z_value < -1.96  :
    print("We Reject the Null Hypothesis(HO)")
else :
    print("We Do Not Reject The Null Hypothesis(HO)")

We Reject the Null Hypothesis(HO)


Hence raw cornstarch will have a effect on blood glucose levels.

## Problem Statement 2:

In one state, 52% of the voters are Republicans, and 48% are Democrats. In a second state, 47% of the voters are Republicans, and 53% are Democrats. Suppose a simple random sample of 100 voters are surveyed from each state. What is the probability that the survey will show a greater percentage of Republican voters in the second state than in the first state?

In [5]:
import numpy as np
from scipy.stats import norm

In [6]:
P1 = 0.52 #proportion of Republican voters in the first state
P2 = 0.47 #the proportion of Republican voters in the second state
n = 100

In [7]:
mean = P1 - P2 #mean of the difference in sample proportions
std = np.sqrt(( P1 * (1 - P1) / n ) + ( P2 * (1 - P2) / n ))

This problem requires us to find the probability that p1 is less than p2. This is equivalent to finding the probability that p1 - p2 is less than zero. To find this probability, we need to transform the random variable (p1 - p2) into a z-score.

In [8]:
z_score = (0-mean)/std
print("Z value :",z_score)

Z value : -0.7080278164104213


In [9]:
prob = norm.cdf(z_score,0,1)

In [10]:
print("The probability that the survey will show a greater percentage of Republican voters in the second state than in the first state is ",prob)

The probability that the survey will show a greater percentage of Republican voters in the second state than in the first state is  0.23946399182220013


## Problem Statement 3:

You take the SAT and score 1100. The mean score for the SAT is 1026 and the standard
deviation is 209. How well did you score on the test compared to the average test taker?

In [11]:
from scipy.stats import norm

In [12]:
mean_score = 1026
std = 209
score = 1100

In [13]:
z_value = (score - mean_score) / std
print("Z-Value : ",z_value)

Z-Value :  0.35406698564593303


This means that your score was 0.35406698564593303 std devs above the mean.

In [14]:
probability = norm.cdf(z_value,0,1)

In [15]:
print("percentage of test-takers scored :",probability,"%")

percentage of test-takers scored : 0.6383556584353189 %
